Topics/Tech_Stacks/Writing_Clean_React_Code.md

```diff
@@ -0,0 +1,153 @@
+
+  ```jsx
+  const Button = ({ text, onClick }) => {
+    return <button onClick={onClick}>{text}</button>;
+  };
+  ```
+
+Notice that the onClick handler isn't defined in the component, and prop name isn't prefixed by _handle_. This is because the component is only responsible for rendering the button and not handling the button click. The onClick handler should be defined in the parent component that uses the Button component, and _in the parent component_ the event handler that is passed down to the Button component should be prefixed by _handle_. Like in the below example:
+
+```jsx
+const ParentComponent = () => {
+  const handleClick = () => {
+    // handle button click
+  };
+  return <Button text="Click Me" onClick={handleClick} />;
+};
+```
+
+- **Constants**: Constants should be named using uppercase letters and underscores to separate words. For example, a constant that stores user emails should be named `USER_EMAILS` instead of `user-emails` or `userEmails`.
+
+- **File Names**: File names should be named using PascalCase. For example, a file that contains a component that renders a navbar element should be named `NavbarElement.jsx` instead of `navbarElement.jsx` or `navbar-element.jsx`.
+
+- **CSS Classes and Files**: CSS classes should be named using kebab-case or camelCase. For example, a class that styles a div that acts as a container for a cards should be named `cards-container` or `cardsContainer` instead of `cards_container`. CSS Files follows the same convention as CSS classes.
+
+As to what you should name your classes, a popular convention is to use the BEM (Block, Element, Modifier) naming convention. This convention is useful for naming classes that are used to style components. It is a pretty big topic in its own right, so to learn more you might want to check out the [getbem.com website](https://getbem.com/introduction/).
+
+## Component Structure
+
+When working with React, it is important to follow a consistent component structure. This helps developers quickly identify the purpose of each component and its role in the application. The following are some common principles for structuring React components:
+
+- **Break down components**: Components should be broken down into smaller components whenever possible. Large components are difficult to understand and debug. It is also easier to reuse smaller components, which avoids code repetition. This essentially goes back to the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) in software engineering, which states that each component should have a single responsibility. For example, a component that renders a user profile page should be broken down into smaller components that render the user's name, profile picture, and bio.
```

Comment:
The definition of the Single Responsibility Principle here is rather redundant; maybe expand on it more, even though you bring up an example next. Specifically, elaborate on what you mean by responsibility when you say that a component should only have a single responsibility.